backend/modules/invoices.js

```diff
@@ -1,132 +1,145 @@
+      if (!req.user) {
+        return res.status(401).json({ success: false, message: 'Unauthorized' });
+      }
+
+      const { id } = req.params;
+      const { eventId } = req.query;
+
+      const data = await pool.query('SELECT * FROM "public"."invoices" WHERE id = $1 AND "eventId" = $2', [id, eventId]);
+
+      if (data.rowCount === 0) {
+        return res.status(404).json({
+          success: false,
+          message: 'Invoice not found',
+        });
+      }
+
+      const invoice = data.rows[0];
+      const file = fs.readFileSync(`invoices/${invoice.fileUrl}`);
+      const mimetype = invoice.fileUrl.split('.').pop();
+      res.setHeader('Content-Type', `image/${mimetype}`);
+      res.send(file);
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ success: false, message: 'Internal Server Error !' });
+    }
+  });
+
 
-const router = Router();
-const app = express();
 
 // Middleware
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: true }));
-app.use(fileUpload());  
-
-// Check if the role is valid (EC, CC, or JC)
-function checkRole(req, res, next) {
-  const { role } = req.query;
-  
-  if (role !== 'EC' && role !== 'CC' && role !== 'JC') {
-    return res.status(403).json({ error: 'Unauthorized: Invalid role' });
-  }
-  next();
-}
+router.use(bodyParser.json());
+router.use(bodyParser.urlencoded({ extended: true }));
 
-// Route to add a new invoice
-router.post('/new', async (req, res) => {
+// Endpoint to add a new invoice
+router.post('/invoice/new', async (req, res) => {
   try {
-    if (!req.user) {
-      return res.status(401).json({ success: false, message: 'Unauthorized' });
-    }
-
-    // Get the file from body
     const { amount, eventId } = req.query;
-    if (!amount || !eventId) {
-      return res.status(400).json({ success: false, message: 'Amount and eventId are required' });
-    }
 
-    if (!req.files || Object.keys(req.files).length === 0) {
-      return res.status(400).json({ success: false, message: 'No file uploaded.' });
+    // amount and eventId are necessary
+    if (!amount || !eventId) {
+      return res.status(400).json({ success: false, message: 'amount and eventId are required in query parameters' });
     }
-    const file = req.files.file;
-    const name = `${req.user.id}-${uuidv4()}-${file.name}`;
-    await file.mv(`invoices/${name}`);
 
+    // Create a new invoice object
     const id = uuidv4();
-    await pool.query('INSERT INTO "public"."invoices" (id, "fileUrl", amount, "createdAt", "createdBy", accepted, "acceptedBy", "acceptedAt", "eventId") ' +
-      'VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)', [id, name, amount, new Date(), req.user.id, false, null, null, eventId]);
+    const status = 'pending';
+
+    // Insert the new invoice into the database
+    const result = await pool.query(
+      'INSERT INTO "public"."invoices" (id, amount, "createdAt", "createdBy", accepted, "acceptedBy", "acceptedAt", eventId) ' +
```

Comment:
`eventId` will come in double quotes, else SQL will lowercase it and throw error.